boards/common/e104-bt50xxa-tb/board.c

```diff
@@ -20,13 +20,22 @@
 
 #include "cpu.h"
 #include "board.h"
+#include "periph/gpio.h"
+
+extern void pm_reboot(void*);
 
 void board_init(void)
 {
     /* initialize the boards LEDs */
     LED_PORT->DIRSET = (LED_MASK);
     LED_PORT->OUTSET = (LED_MASK);
 
+    /* configure software RST button */
+    if (IS_ACTIVE(MODULE_BOARD_SOFTWARE_RESET)) {
+        gpio_init_int(BTN0_PIN, BTN0_MODE, GPIO_FALLING,
+                      pm_reboot, NULL);
+    }
+
```

Comment:
`DISABLE_MODULE=board_software_reset BOARD=e104-bt5011a-tb make -C examples/hello-world` yields a compilation error since `periph_gpio_irq` is not included. So I think you will have to use `ifdef`